src/MongoDB.Driver/UpdateDefinition.cs

```diff
@@ -66,6 +79,128 @@ public abstract class UpdateDefinition<TDocument>
+        {
+            if (pipeline == null)
+            {
+                return null;
+            }
+            return new PipelineUpdateDefinition<TDocument>(pipeline);
+        }
+    }
+
+    /// <summary>
+    /// A <see cref="PipelineDefinition{TDocument,TDocument}"/> based update definition.
+    /// </summary>
+    /// <typeparam name="TDocument">The type of the document.</typeparam>
+    public sealed class PipelineUpdateDefinition<TDocument> : UpdateDefinition<TDocument>
+    {
+        private readonly PipelineDefinition<TDocument, TDocument> _pipeline;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PipelineUpdateDefinition{TDocument}"/> class.
+        /// </summary>
+        /// <param name="pipeline">The pipeline.</param>
+        public PipelineUpdateDefinition(PipelineDefinition<TDocument, TDocument> pipeline)
+        {
+            _pipeline = Ensure.IsNotNull(pipeline, nameof(pipeline));
+        }
+
+        /// <summary>
+        /// Renders the update to a <see cref="BsonValue"/> that represents <see cref="BsonArray"/>.
+        /// </summary>
+        /// <param name="documentSerializer">The document serializer.</param>
+        /// <param name="serializerRegistry">The serializer registry.</param>
+        /// <returns>A <see cref="BsonValue"/>.</returns>
+        public override BsonValue Render(IBsonSerializer<TDocument> documentSerializer, IBsonSerializerRegistry serializerRegistry)
+        {
+            var renderedPipeline = _pipeline.Render(documentSerializer, serializerRegistry);
+            return new BsonArray(renderedPipeline.Documents);
+        }
+
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var serializerRegistry = BsonSerializer.SerializerRegistry;
+            var inputSerializer = serializerRegistry.GetSerializer<TDocument>();
+            return ToString(inputSerializer, serializerRegistry);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="System.String" /> that represents this instance.
+        /// </summary>
+        /// <param name="inputSerializer">The input serializer.</param>
+        /// <param name="serializerRegistry">The serializer registry.</param>
+        /// <returns>
+        /// A <see cref="System.String" /> that represents this instance.
+        /// </returns>
+        public string ToString(IBsonSerializer<TDocument> inputSerializer, IBsonSerializerRegistry serializerRegistry)
+        {
+            var renderedPipeline = Render(inputSerializer, serializerRegistry);
+            return renderedPipeline.ToJson();
+        }
+
+        /// <summary>
+        /// Creates an update pipeline.
+        /// </summary>
+        /// <param name="pipeline">The pipeline stages.</param>
+        /// <returns>A <see cref="PipelineUpdateDefinition{TDocument}"/>.</returns>
+        public static PipelineUpdateDefinition<TDocument> Create(BsonArray pipeline)
```

Comment:
I would remove the two Create methods and the two implicit operators. They are not really needed.